### PR TITLE
Update print_variable functions and test_output files to comply with printf output.

### LINKED ISSUE #32: Invalid scientific notation output by `print_state()`
### Issue checklist

- [x] I have read the [README](https://github.com/NOAA-GSL/SENA-c_sw/blob/develop/README.md) and [ref/README](https://github.com/NOAA-GSL/SENA-c_sw/blob/develop/ref/README.md), including the Troubleshooting Guide.
- [x] I have reviewed existing issues to ensure this has not already been reported.

### Describe the bug

The `print_state()` function is outputting non-standard scientific notation, and incompatible with standards used by other languages (C, C++, Julia) and other `printf` based I/O.

### Recipe to reproduce

1. Go to the print_state() function in  [sw_core.F90](https://github.com/NOAA-GSL/SENA-c_sw/blob/develop/ref/src/sw_core.F90) 
2. Observe `print_state()`'s calls to `print_3d_variable()` and `print_2d_variable()` functions. These functions' `write` statement should be altered to comply with the 'conventional' way that `printf` represents floating point numbers in scientific notation. 

### Expected behavior, Screenshots, and terminal output

The current output in scientific notation:

```-0.10000000000E-07```

`printf`'s equivalent of how this value is expressed in scientific notation:

```-1.00000000000E-08```

This is what the modified Fortran code should output.

 See [current test outputs](https://github.com/NOAA-GSL/SENA-c_sw/blob/develop/ref/test/test_output/c_sw_12x24.test) for full context. 

### Additional context

Fortran is the outlier in how scientific notation is expressed. The current output will make it very difficult for others to use baselines for testing. 

Not only will the `write` statements in the `print_3d_variable` & `print_2d_variable` functions need to be changed, all of the files within the [ref/test/test_output](https://github.com/NOAA-GSL/SENA-c_sw/tree/develop/ref/test/test_output) will need alterations to reflect the new scientific notation necessary to comply with future testing of the kernel.

### PR DESCRIPTION
**Pull request by robgpita**
## Proposed changes

This PR closes #32. The `print_state()` function is outputting non-standard scientific notation, and incompatible with standards used by other languages (C, C++, Julia) and other `printf` based I/O. 

Fortran is the outlier in how scientific notation is expressed. The current output makes it very difficult for others to use baselines for testing. 

The `write` statements in the `print_3d_variable` & `print_2d_variable` functions have been changed, and all of the files within the [ref/test/test_output](https://github.com/NOAA-GSL/SENA-c_sw/tree/develop/ref/test/test_output) directory have been altered to reflect the new scientific notation necessary to comply with future testing of the kernel. 

## Types of changes

What types of changes does your code introduce?

- [x] Bugfix (Change which fixes an issue)
- [ ] New feature (Change which adds functionality)
- [ ] Enhancement to existing functionality (non-breaking change which improves existing code or documentation)
- [x] Breaking change (fix or feature that changes results and requires updates to test baselines)
- [ ] Documentation Update

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/NOAA-GSL/SENA-c_sw/blob/develop/CONTRIBUTING.md) document
- [x] I have ensured all my changes contribute to a common purpose
- [x] I have verified my change does not add debug code
- [x] I have verified my change does not add code that is commented out
- [x] I have verified unit tests pass locally with my changes
- [x] I have verified my changes adhere to style guidelines
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

### CODE REVIEW

**Review by middlecoff — DISMISSED** [review]
Looks good.

File: ref/test/test_output/c_sw_12x24.test [hunk 1]
```diff
@@ -3,82 +3,82 @@ TEST          Input State
 TEST ===================================================================================================================
 TEST        Variable                 Min                 Max               First                Last                 RMS
 TEST -------------------------------------------------------------------------------------------------------------------
-TEST           rarea  -0.10000000000E-07   0.68896173601E-09  -0.10000000000E-07   0.56546800912E-09   0.14365610413E-08
-TEST         rarea_c   0.56234680356E-09   0.69052250213E-09   0.67316411530E-09   0.56234680356E-09   0.63558157249E-09
```
Comment (LyndStringer) [inline ref/test/test_output/c_sw_12x24.test:7]: Ouch, this PR increases the accuracy from 11 digits to 12 digits.  When I added interpolation to SENA-c_sw (not a PR yet) I was having troubling getting 11 digits of accuracy.  12 digits will be troublesome.

**Review by LyndStringer — CHANGES_REQUESTED** [review]
At a minimum, we need to talk about the 11 to 12 digits change.  We have already throttled back the fused multiply-adds to keep 11 digits.  Again, 12 digits will be troublesome.

File: ref/test/test_output/c_sw_12x24.test (hunk 1 above)
Comment (christopherwharrop-noaa) [reply ref/test/test_output/c_sw_12x24.test:7]: The increase was unintentional.  @robgonzalezpita, could you please change the `5ES20.11` to `5ES20.10` and regenerate new baselines?  @LyndStringer is correct.

Comment (robgpita) [reply ref/test/test_output/c_sw_12x24.test:7]: @christopherwharrop-noaa, you're right, the increased accuracy was indeed unintentional. I created this PR knowing there would be some necessary adjustments. @LyndStringer,  I'll make the changes here.

**Review by LyndStringer — APPROVED** [review]
Looks good!

**Pull request merged**